habitat/sims/pyrobot/pyrobot.py

```diff
@@ -0,0 +1,281 @@
+class PyRobot(Simulator):
+    r"""Simulator wrapper over PyRobot.
+
+    PyRobot repo: https://github.com/facebookresearch/pyrobot
+    To use this abstraction the user will have to setup PyRobot
+    python3 version. Please refer to the PyRobot repository
+    for setting it up. The user will also have to export a
+    ROS_PATH environment variable to use this integration,
+    please refer to `habitat.core.utils.try_cv2_import` for
+    more details on this.
+
+    This abstraction assumes that reality is a simulation
+    (https://www.youtube.com/watch?v=tlTKTTt47WE).
+
+    Args:
+        config: configuration for initializing the PyRobot object.
+    """
+
+    def __init__(self, config: Config) -> None:
+        self._config = config
+
+        robot_sensors = []
+        for sensor_name in self._config.SENSORS:
+            sensor_cfg = getattr(self._config, sensor_name)
+            sensor_type = registry.get_sensor(sensor_cfg.TYPE)
+
+            assert sensor_type is not None, "invalid sensor type {}".format(
+                sensor_cfg.TYPE
+            )
+            robot_sensors.append(sensor_type(sensor_cfg))
+        self._sensor_suite = SensorSuite(robot_sensors)
+
+        config_pyrobot = {
+            "base_controller": self._config.BASE_CONTROLLER,
+            "base_planner": self._config.BASE_PLANNER,
+        }
+
+        assert (
+            self._config.ROBOT in self._config.ROBOTS
+        ), "Invalid robot type {}".format(self._config.ROBOT)
+        self._robot_config = getattr(self._config, self._config.ROBOT.upper())
+
+        action_spaces_dict = {}
+
+        self._action_space = self._robot_action_space(
+            self._config.ROBOT, self._robot_config
+        )
+
+        self._robot = pyrobot.Robot(
+            self._config.ROBOT, base_config=config_pyrobot
+        )
+
+    def get_robot_observations(self):
+        return {
+            "rgb": self._robot.camera.get_rgb(),
+            "depth": self._robot.camera.get_depth(),
+            "bump": self._robot.base.base_state.bumper,
+        }
+
+    @property
+    def sensor_suite(self) -> SensorSuite:
+        return self._sensor_suite
+
+    @property
+    def base(self):
+        return self._robot.base
+
+    @property
+    def camera(self):
+        return self._robot.camera
+
+    def _robot_action_space(self, robot_type, robot_config):
+        action_spaces_dict = {}
+        for action in robot_config.ACTIONS:
+            action_spaces_dict[action] = ACTION_SPACES[robot_type.upper()][
+                action
+            ]
+        return spaces.Dict(action_spaces_dict)
+
+    @property
+    def action_space(self) -> Space:
+        return self._action_space
+
+    def reset(self):
+        self._robot.camera.reset()
+
+        robot_obs = self.get_robot_observations()
+        observations = self._sensor_suite.get_observations(robot_obs)
+        return observations
+
+    def step(self, action, action_params):
```

Comment:
Maybe still have default for action without params.
```suggestion
    def step(self, action, action_params=None):
```
Overall, it looks expected `Simulator.step` follow `Task.step` interfaces with `step(action: Dict,...)`.